zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -345,16 +370,20 @@ public Response createNote(String message) throws IOException {
       }
     }
     note.addNewParagraph(subject); // add one paragraph to the last
-    String noteName = request.getName();
-    if (noteName.isEmpty()) {
-      noteName = "Note " + note.getId();
+
+    //deal with possible empty note name
+    String noteName = "Note " + note.getId();
+    if (request != null && request.getName() != null) {
+      if (!request.getName().isEmpty()) {
+        noteName = request.getName();
+      }
     }
 
     note.setName(noteName);
     note.persist(subject);
     notebookServer.broadcastNote(note);
     notebookServer.broadcastNoteList(subject, SecurityUtils.getRoles());
-    return new JsonResponse<>(Status.OK, "", note.getId()).build();
+    return new JsonResponse<>(Status.CREATED, "", note.getId()).build();
```

Comment:
if I read this right, the doc says it should be 200 OK
https://github.com/apache/zeppelin/pull/2508/files#diff-ccd7e2a2e181424ea2f4e346189e7688R93
?